Lib/os.py

```diff
@@ -1136,3 +1136,16 @@ def add_dll_directory(path):
             cookie,
             nt._remove_dll_directory
         )
+
+
+if _exists('sched_getaffinity'):
+    def process_cpu_count():
+        """
+        Get the number of logical CPUs usable by the current process.
+
+        Return None if indeterminable.
+        """
+        return len(sched_getaffinity(0))
```

Comment:
Could this ever return an empty set for any reason?  it might be safer to do something like

```python
cpu_set = sched_getaffinity(0)
num_cpus = cpu_count()
return min(len(cpu_set), num_cpus) if cpu_set else num_cpus
```

Then again, I'm probably overthinking things.  The system API should never return an empty set.

Reply:
_(mostly wondering if an empty set signifies "all" on some systems)_

Reply:
On Linux, you cannot assign 0 CPUs to a process:

```
$ python
>>> import os
>>> os.sched_setaffinity(0, [])
OSError: [Errno 22] Invalid argument
```

man sched_getaffinity:

```
ERRORS
       EINVAL The affinity bit mask mask contains no proces-
              sors that are currently physically on the sys-
              tem  and  permitted to the thread according to
              any restrictions that may be imposed by cpuset
              cgroups or the "cpuset" mechanism described in
              cpuset(7).

       EINVAL (sched_getaffinity() and, before Linux  2.6.9,
              sched_setaffinity())   cpusetsize  is  smaller
              than the size of the affinity mask used by the
              kernel.
```

From what I understand, sched_setaffinity() cannot set an empty list of CPU.

Also os.sched_getaffinity() **cannot fail**. sched_getaffinity() returns EINVAL if the mask is too small, Python uses a loop of make this list larger until it's big enough.

Reply:
After the discussion with @vstinner, I concur his comment.